Automattic/woocommerce-payments
# Add Stripe's Payment Method Messaging Element to checkout radio buttons

Fixes #8143

#### Changes proposed in this Pull Request

This PR adds the PMME to both the blocks and classic checkout. I've done my best to introduce a base CSS style for aligning this new information according to the figma design, but fundamental differences in how themes handle rendering the radio input results in some variations from theme to theme. Some themes hide the radio input inside the payment method list item and use another method to make a fake radio button (like style a `:before` element or style the disc from the `li`). These differences are most apparent on the classic checkout.

<img width="200" alt="Screenshot 2024-04-04 at 10 16 02 AM" src="https://github.com/Automattic/woocommerce-payments/assets/1558827/8e914116-06e6-406d-8669-0c155a306a18"> <img width="200" alt="Screenshot 2024-04-04 at 10 16 11 AM" src="https://github.com/Automattic/woocommerce-payments/assets/1558827/940e1bfd-e5c2-4a12-8741-de346c19c9f1"> <img width="200" alt="Screenshot 2024-04-04 at 10 16 21 AM" src="https://github.com/Automattic/woocommerce-payments/assets/1558827/5022563c-8fc7-46e7-9db3-d56b1787612e">

#### Testing instructions

1. Enable BNPL methods, i.e. Affirm, After Pay, Klarna
2. Add a product to the cart
3. Ensure that the shopper currency, country, and total amounts align with the requirements for BNPL payment methods. Each set of requirements can be found in the payment methods class. For example, the Klarna requirements can be found [here](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/payment-methods/class-klarna-payment-method.php#L34-L102).
4. Check that the BNPL text is rendered under the title of the BNPL method on both the classic checkout and block checkout and that the displayed amount updates when cart totals change(shipping method change or coupons, etc.).
5. Also test with different themes to see if this information ever looks "broken."

-------------------


**Post merge**
